src/services/createPropStore.ts

```diff
@@ -2,43 +2,64 @@ import { InjectionKey, reactive } from 'vue'
 import { isWithComponent, isWithComponents } from '@/types/createRouteOptions'
 import { ResolvedRoute } from '@/types/resolved'
 import { Route } from '@/types/route'
-import { MaybePromise } from '@/types/utilities'
 
 export const propStoreKey: InjectionKey<PropStore> = Symbol()
 
 type ComponentProps = { id: string, name: string, props?: (params: Record<string, unknown>) => unknown }
 
 export type PropStore = {
-  setProps: (route: ResolvedRoute) => void,
-  getProps: (id: string, name: string, params: unknown) => MaybePromise<unknown> | undefined,
+  setProps: (route: ResolvedRoute, options: { prefetched: boolean }) => void,
+  getProps: (id: string, name: string, params: Record<string, unknown>) => unknown,
 }
 
 export function createPropStore(): PropStore {
-  const store = reactive(new Map<string, unknown>())
+  const store: Map<string, unknown> = reactive(new Map())
```

Comment:
`reactive` produces a mapped type because of ref unwrapping. Writing this way produces a simpler type for the `store`.